orchestration-shared/src/main/java/uk/gov/di/orchestration/shared/entity/VectorOfTrust.java

```diff
@@ -46,132 +35,36 @@ public LevelOfConfidence getLevelOfConfidence() {
-    }
-
-    public String retrieveVectorOfTrustForToken() {
-        return credentialTrustLevel.getValue();
-    }
-
-    public boolean isValid() {
-        return nonNull(getCredentialTrustLevel())
-                && !(Objects.equals(getLevelOfConfidence(), LevelOfConfidence.MEDIUM_LEVEL)
-                        && Objects.equals(
-                                getCredentialTrustLevel(), CredentialTrustLevel.LOW_LEVEL));
-    }
-
-    private static List<VectorOfTrust> parseVtrSet(JSONArray vtrJsonArray) {
-        List<VectorOfTrust> vtrList = new ArrayList<>();
-        for (Object obj : vtrJsonArray) {
-            String vtr = (String) obj;
-            var splitVtr = vtr.split("\\.");
-
-            var levelOfConfidence =
-                    Arrays.stream(splitVtr)
-                            .filter(a -> a.startsWith("P"))
-                            .map(LevelOfConfidence::retrieveLevelOfConfidence)
-                            .collect(
-                                    Collectors.collectingAndThen(
-                                            Collectors.toList(),
-                                            list -> {
-                                                if (list.size() > 1) {
-                                                    throw new IllegalArgumentException(
-                                                            "VTR must contain either 0 or 1 identity proofing components");
-                                                }
-                                                return list;
-                                            }))
-                            .stream()
-                            .findFirst();
-
-            var credentialTrustLevel =
-                    CredentialTrustLevel.retrieveCredentialTrustLevel(
-                            Arrays.stream(splitVtr)
-                                    .filter(a -> a.startsWith("C"))
-                                    .sorted()
-                                    .collect(Collectors.joining(".")));
-            var vot = new VectorOfTrust(credentialTrustLevel, levelOfConfidence);
-            if (!vot.isValid()) {
-                throw new IllegalArgumentException(
-                        "P2 identity confidence must require at least Cl.Cm credential trust");
-            }
-            vtrList.add(vot);
-        }
-
-        var identityVectorsCount =
-                vtrList.stream()
-                        .filter(
-                                vtr ->
-                                        Objects.nonNull(vtr.getLevelOfConfidence())
-                                                && !vtr.getLevelOfConfidence().equals(NONE))
-                        .count();
-        if (identityVectorsCount != 0 && identityVectorsCount < vtrList.size()) {
-            throw new IllegalArgumentException(
-                    "VTR cannot contain both identity and non-identity vectors");
-        }
-
-        return vtrList;
```

Comment:
As far as I can tell this exception is thrown in order to make it possible to have the safe assumption later that the we can just look at the first element of the VTR list. I assume this was done for convenience as it would be strange to mix identity and non identity VoTs in a VTR, but given there's a VtrList class now, it doesn't add any convenience, so I didn't see any justification in rejecting it. @ethanmills perhaps I'm wrong and we should continue to reject this kind of VTR? I'd say it should go in a separate PR if it was less permissive, but given it's more permissive I'm not sure it matters.

Reply:
In the interest of small, atomic changes, can we leave for another PR where we can test it separately?

Reply:
I disagree that we should just skim over a change that is more permissive also - the validation is there for a reason, and we should fully scrutinise any change to how permissive a function is

Reply:
No, this change is wrong Ben. What should Orchestration do if it receives a mix of identity and non-identity VTRs? Should it send the user on an identity journey or not? Is a failure in identity a failure or not? It's fundamentally opposed to the way the entire service's logic is put together, and is an illegal state.

Reply:
Okay, thanks, I'll re-introduce it. I understood we could simply not go on an auth only journey ([https://datatracker.ietf.org/doc/html/rfc8485#section-4.1](https://datatracker.ietf.org/doc/html/rfc8485#section-4.1)).

Reply:
I've re-added the check, but it doesn't look like we document that we reject mixed VTRs.

- [https://docs.sign-in.service.gov.uk/before-integrating/choose-the-level-of-identity-confidence/#choose-the-level-of-identity-confidence-for-your-service](https://docs.sign-in.service.gov.uk/before-integrating/choose-the-level-of-identity-confidence/#choose-the-level-of-identity-confidence-for-your-service)
- [https://docs.sign-in.service.gov.uk/integrate-with-integration-environment/authenticate-your-user/#replace-the-placeholder-values-in-your-example:~:text=The%20vtr%20parameter,in%20your%20response](https://docs.sign-in.service.gov.uk/integrate-with-integration-environment/authenticate-your-user/#replace-the-placeholder-values-in-your-example:~:text=The%20vtr%20parameter,in%20your%20response).

Just to expand a little on the quick discussion we had yesterday @ethanmills re this, given we've actively chosen to have a P0 level rather than simply skipping IPV if we don't see a P-value, doesn't that lend itself to RPs seeing some degree of equivalence between P0 and the other levels. They shouldn't really have to care about how one login is structured internally and I could see an RP writing something like:
`vtr = [new vot("Cl.Cm", "P2")]
if (some_condition) {
    vtr.add(new vot("Cl.Cm", LOW_P_VALUE))
}`
where "LOW_P_VALUE" could be P1 or P0
This makes even more sense given the recent decision not to allow a P2 from IPV against a P1 request.

The other side of this is, if we're going to make a guess that redundancy in the VTR means the RP has made an unintended request, shouldn't we also watch out for duplication, ["Cl", "Cl.Cm"], or a P1 without a P2? I would have thought we should just accept any valid request according to the spec and if we have any concerns that RPs are using the API incorrectly because of redundant VoTs in requests, do some auditing to spot that and contact them.

Personally I don't think it makes much sense to keep this check, particularly given going out of spec with Cl.Cm not actually being a composition of Cl and CM caused confusion with  the RPs but at the very least is it not worth acknowledging this in the documentation?

Reply:
After discussing we've concluded we should maintain the old behaviour but add some documentation. Ticket for documentation added to backlog under P1 epic as only when that is live will there be any reason for most RPs to send multi value VTRs.